### PR TITLE
chore: enable building kythe extractor docker image via GCB

### PR DESCRIPTION
**Pull request by jaysachs**
Haven't tried this out yet, looking for feedback on the approach

### CODE REVIEW

File: kythe/extractors/bazel/gcb.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - '--stamp'
+      - '//kythe/extractors/bazel:stage'
+  - name: gcr.io/cloud-builders/docker
+    id: docker-build
+    args:
+      - build
+      - '-t'
+      - gcr.io/kythe-public/bazel-extractor:latest
+      - .
+    dir: code/bazel-out/k8-opt/bin/kythe/extractors/bazel/stage.done
```
Comment (jaysachs) [inline kythe/extractors/bazel/gcb.yaml:37]: Felt dirty hardcoding this.

Comment (shahms) [reply kythe/extractors/bazel/gcb.yaml:36]: I don't know of any other way, unfortunately.

**Comment by justbuchanan** [comment]
Overall looks good. A couple questions:

What do we gain by running the `docker build` step outside of bazel? Is this related to the glibc version issue?

Does the image built in the docker-build step still exist when the docker-push step runs? Data persistence across build steps is fairly limited (mostly to /workspace or custom volumes), but maybe there's something special for docker images. See also https://cloud.google.com/build/docs/configuring-builds/pass-data-between-steps#passing_data_using_workspaces

**Comment by justbuchanan** [comment]
> What do we gain by running the `docker build` step outside of bazel? Is this related to the glibc version issue?

Following up on my own question - is the issue that the docker binary isn't present in the bazel image, so we're using a separate image with docker installed to do the actual `docker build` part?

**Comment by jaysachs** [comment]
With the latest commit ([cdee542](https://github.com/kythe/kythe/pull/5510/commits/cdee542c0a46a95bdf5d003d3094a623ca22ce0d))  and some GCP IAM fiddling this successfully builds and pushes.

**Review by shahms — APPROVED** [review]

**Pull request merged**